src/com/google/javascript/jscomp/ProductionCoverageInstrumentationCallback.java

```diff
@@ -137,56 +137,123 @@ private Node newInstrumentationNode(NodeTraversal traversal, Node node, String f
     return exprNode.useSourceInfoIfMissingFromForTree(node);
   }
 
+
+  public VariableMap getParamMapping() {
+    return parameterMapping.getParamMappingAsVariableMap();
+  }
+
   /**
    * A class the maintains a mapping of unique identifiers to parameter values. It also generates
    * unique identifiers by creating a counter starting form 0 and increments this value when
-   * assigning a new unique identifier.
+   * assigning a new unique identifier. It converts the mapping to a VariableMap as well so that it
+   * can later be saved as a file.
    */
   private static final class ParameterMapping {
 
-    private final List<String> uniqueIdentifier;
-    private final List<String> paramValue;
+    // Values are stored as a mapping of the String of the encoded array indices to a String of the
+    // encoded uniqueIdentifier. This is so we can check if an encoded param has already been
+    // defined so that we do not create a duplicate. (Ex. Key: ACA (Base64 VLQ encoding of [0,1,0]),
+    // Value: C (Base64 VLQ encoding of the uniqueIdentifier). This map will later be inversed so
+    // that it is printed in the following form: C:ACA.
+    private final Map<String, String> paramValueEncodings;
+
+    // Values are stored as a mapping of String to Integers so that we can lookup the index of the
+    // encoded (file|function|type) name and also check if it is present in constant time. These
+    // mappings are added to the VariableMap once instrumentation is complete.
+    private final Map<String, Integer> fileNamesMap;
```

Comment:
I think names like `fileNameToIndex` would be clearer. It makes it easier to understand what the values are in this map without having to go back o this comment.